mobile/src/main/java/ch/epfl/sdp/mobile/infrastructure/speech/android/AndroidSpeechRecognizerFactory.kt

```diff
@@ -0,0 +1,72 @@
+package ch.epfl.sdp.mobile.infrastructure.speech.android
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.speech.RecognizerIntent.*
+import android.speech.SpeechRecognizer as NativeSpeechRecognizer
+import android.speech.SpeechRecognizer.RESULTS_RECOGNITION
+import ch.epfl.sdp.mobile.infrastructure.speech.SpeechRecognizer
+import ch.epfl.sdp.mobile.infrastructure.speech.SpeechRecognizerFactory
+
+/** The default locale we'll be using for speech recognition. */
+private const val DefaultLanguage = "en-US"
+
+/** The default maximum number of results we'll receive from the system speech recognizer. */
+private const val DefaultResultsCount = 10
+
+/**
+ * An implementation of a [SpeechRecognizer] which is backed by a [NativeSpeechRecognizer].
+ *
+ * @param context the [Context] which will be used to create the speech recognizers.
+ * @param language the default language to use.
+ * @param resultsCount the maximum results count for each result set.
+ */
+class AndroidSpeechRecognizerFactory(
+    private val context: Context,
+    private val language: String = DefaultLanguage,
+    private val resultsCount: Int = DefaultResultsCount,
+) : SpeechRecognizerFactory {
```

Comment:
[Question] : Having a factory allow us to a different recognizer that understand multiple languages, right ?

Reply:
We could eventually provide a way to create a `SpeechRecognizer` which recognises a specific language by providing some parameter to `SpeechRecognizerFactory.createSpeechRecognizer(language: String = DefaultLanguage)`, but I wasn't quite sure how we'd handle languages in the rest of the text processing pipeline so I haven't considered it further.

The code somehow mimics what had been done in #201.